mempool/iterators.go

```diff
@@ -100,18 +101,22 @@ func (iter *NonBlockingIterator) Next() Entry {
 // Unlike `NonBlockingIterator`, this iterator is expected to work with an evolving mempool.
 type BlockingIterator struct {
 	IWRRIterator
-	mp *CListMempool
+	ctx  context.Context
+	mp   *CListMempool
+	name string // for debugging
 }
 
-func NewBlockingIterator(mem *CListMempool) Iterator {
+func NewBlockingIterator(ctx context.Context, mem *CListMempool, name string) Iterator {
 	iter := IWRRIterator{
 		sortedLanes: mem.sortedLanes,
 		cursors:     make(map[types.Lane]*clist.CElement, len(mem.sortedLanes)),
 		round:       1,
 	}
 	return &BlockingIterator{
 		IWRRIterator: iter,
+		ctx:          ctx,
 		mp:           mem,
+		name:         name,
```

Comment:
If the non blocking iterator also has a name, I would prefix this with "B" and the other one with "NB". But not mandatory.

Reply:
I don't think we need a name for the non-blocking iterator. Usually there's only one active non-blocking iterator in the mempool (when reaping). When rechecking there's one for sending the CheckTx request and another for processing the responses. In all those cases, the mempool doesn't change. It's different for blocking iterators because there could be many acting concurrently and with a dynamic mempool (for example, the name is used in `TestBlockingIteratorsConsumeAllTxs`).